oap-server/server-storage-plugin/storage-elasticsearch-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/elasticsearch/base/IndexController.java

```diff
@@ -91,12 +109,42 @@ public static String getPhysicalTableName(String logicName) {
             return Optional.ofNullable(LOGIC_INDICES_CATALOG.get(logicName)).orElse(logicName);
         }
 
-        public static void registerRelation(String logicName, String physicalName) {
-            LOGIC_INDICES_CATALOG.put(logicName, physicalName);
+        public static void registerRelation(Model model, String physicalName) {
+            LOGIC_INDICES_CATALOG.put(model.getName(), physicalName);
+            Map<String, ModelColumn> columns = PHYSICAL_INDICES_COLUMNS.computeIfAbsent(
+                physicalName, v -> new ConcurrentHashMap<>());
+            model.getColumns().forEach(modelColumn -> {
+                String columnName = modelColumn.getColumnName().getName();
+                if (columns.containsKey(columnName)) {
+                    checkModelColumnConflicts(columns.get(columnName), modelColumn, physicalName);
+                } else {
+                    columns.put(columnName, modelColumn);
+                }
+            });
         }
 
-        public static boolean isMetricTable(String logicName) {
+        public static boolean isPhysicalTable(String logicName) {
             return !getPhysicalTableName(logicName).equals(logicName);
         }
+
+        public static List<ModelColumn> getPhysicalTableColumns(Model model) {
+            String tableName = getPhysicalTableName(model.getName());
+            return new ArrayList<>(PHYSICAL_INDICES_COLUMNS.get(tableName).values());
+        }
+
+        private static void checkModelColumnConflicts(ModelColumn mc1, ModelColumn mc2, String physicalName) {
+            if (!(mc1.isIndexOnly() == mc2.isIndexOnly())) {
+                throw new IllegalArgumentException(mc1.getColumnName() + " and " + mc2.getColumnName() + " isIndexOnly conflict in index: " + physicalName);
+            }
+            if (!(mc1.isStorageOnly() == mc2.isStorageOnly())) {
+                throw new IllegalArgumentException(mc1.getColumnName() + " and " + mc2.getColumnName() + " isStorageOnly conflict in index: " + physicalName);
+            }
```

Comment:
For these conflicts, we should indicate the module name of new appending column.

Reply:
`getColumnName()` provide full name, including model name